pentasharp/Services/DriverService.cs

```diff
@@ -0,0 +1,109 @@
+﻿using pentasharp.Data;
+using pentasharp.Models.Entities;
+using pentasharp.ViewModel.Authenticate;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using System.Text;
+using System.Security.Cryptography;
+using pentasharp.Models.Enums;
+using pentasharp.ViewModel.TaxiModels;
+using pentasharp.Models.TaxiRequest;
+
+namespace pentasharp.Services
+{
+    public class DriverService : IDriverService
+    {
+        private readonly AppDbContext _context;
+        private readonly IMapper _mapper;
+
+        public DriverService(AppDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task<User> AddDriverAsync(RegisterDriverRequest model, int companyId)
+        {
+            var newUser = new User
+            {
+                FirstName = model.FirstName,
+                LastName = model.LastName,
+                Email = model.Email,
+                PasswordHash = HashPassword(model.Password),
+                CompanyId = companyId,
+                Role = UserRole.Driver,
+                IsAdmin = false,
+                BusinessType = BusinessType.TaxiCompany
+            };
+
+            _context.Users.Add(newUser);
+            await _context.SaveChangesAsync();
+            return newUser;
+        }
+
+        public List<DriverRequest> GetDrivers(int companyId)
+        {
+            var drivers = (from u in _context.Users
+                           join c in _context.TaxiCompanies on u.CompanyId equals c.TaxiCompanyId
+                           where u.CompanyId == companyId
```

Comment:
this query can be improved, one thing that could help is to execute where clause before join as it will reduce the number of rows being joined